### ISSUE TITLE
Get City and Country

### ISSUE THREAD

**Issue by webing3** [tomik23/Leaflet.Autocomplete#5]
@tomik23 hi ! can i get city and county from the results??

**Comment by tomik23** [comment]
Hi,

It depends on what api you are using?
In my example, nominatim is used so I have to build the query properly first, take a [look documentation](https://nominatim.org/release-docs/latest/api/Search/) and then change them in this [line](https://github.com/tomik23/Leaflet.Autocomplete/blob/master/docs/script.js#L11) and also [this line](https://github.com/tomik23/Leaflet.Autocomplete/blob/master/docs/script.js#L69).

I warn you, nominatim returns, more than we would like ;)
Even this working example of mine does not only return cities, but also the names of villages, neighborhoods, etc. ;)

See how geojson is returned https://nominatim.openstreetmap.org/search?format=geojson&limit=5&addressdetails=1&city=sydney
As you can see, these are not just returned cities. It may happen that two identical results will appear in the search results, e.g .:
- Sydney, Australia
- Sydney, Australia

And the question is, what will you do then :)
You can set everything up in [line 69](https://github.com/tomik23/Leaflet.Autocomplete/blob/master/docs/script.js#L69) and allow only those results that are suitable for you.

It is not without reason that the website https://www.openstreetmap.org/search?query=sydney#map=10/-33.7697/150.8024 displays not only the city and country but the entire name.

To sum up, it is possible to display only the city and country, but you have to correctly ask api nominatim how to use the conditions to display the records.

**Comment by jasomdotnet** [comment]
My code for somebody with same request as @shykhamza 

```
function googleGeocoder(string $address, string $lang): array {
        
        $json['features'] = [];

        $url = 'https://maps.googleapis.com/maps/api/geocode/json?' . http_build_query(['address' => $address, 'language' => $lang, 'sensor' => 'false', 'key' => MAP_GOOGLE_APIKEY]);

        $request = json_decode(file_get_contents($url));

        if ($request->status === 'OK') {

            foreach ($request->results as $result) {

                $point = null;

                if (!empty($result->address_components)) {

                    $address_components = [];

                    foreach ($result->address_components as $address_component) {

                        if (!in_array('postal_code', $address_component->types)) {

                            $address_components[] = $address_component->long_name;
                        }
                    }

                    $locality = htmlspecialchars(implode(', ', array_unique($address_components)), ENT_QUOTES, 'UTF-8');

                    $point['properties']['display_name'] = $locality;
                }

                if (!empty($result->geometry->location)) {

                    $point['geometry']['coordinates'] = [$result->geometry->location->lng, $result->geometry->location->lat];
                }

                if (!empty($point)) {

                    $json['features'][] = $point;
                }
            }
        } elseif ($request->status === 'ZERO_RESULTS') {

            //$json['features'] = [];
        } else {

            $log = sprintf('Error within Google Geocoder API with status %1$s for request %2$s.', $request->status, str_replace(MAP_GOOGLE_APIKEY, 'key', $url));

            if (!empty($request->error_message)) {

                $log .= sprintf(' Error message: %s', $request->error_message);
            }

            Log::watchdog('error', $log);
        }
        
        return $json;
    }
```
